### ISSUE TITLE
Document the contract around `provider.proto`'s `rpc Read`

### ISSUE THREAD

**Issue by iwahbe** [pulumi/pulumi#17650]
The entirety of the source documentation for `Read` is:

https://github.com/pulumi/pulumi/blob/2aad59df18f6dd0c27fe2dfef2ab68dd8a5b5f8e/proto/pulumi/provider.proto#L82-L84

https://github.com/pulumi/pulumi/blob/2aad59df18f6dd0c27fe2dfef2ab68dd8a5b5f8e/proto/pulumi/provider.proto#L370-L383

Our [developer docs](https://pulumi-developer-docs.readthedocs.io/latest/docs/_generated/proto/provider.html#read) don't include additional information.

Before we can begin to write comprehensive tests for pulumi-terraform-bridge to validate the bridge against the spec, we need to be clear what the spec is.

**Comment by t0yv0** [comment]
https://github.com/pulumi/pulumi/issues/17656

**Comment by t0yv0** [comment]
Refresh is very under-documented as well especially post https://github.com/pulumi/pulumi/pull/16146 changes. This might deserve breaking out into a separate task. I'd love to have some docu that makes it easier to reason through.

Import (both regular and via import option) is documented in https://pulumi-developer-docs.readthedocs.io/latest/developer-docs/architecture/import.html ; we might need to take a pass and see if there's any remaining questions that need clarifying. Also it's a tad confusing to me why this import page is not find-able from the docs TOC.

**Comment by t0yv0** [comment]
I think Get functions, refresh and the two kinds of import are all the use cases I know so far for the Read method. There is also an emerging resource discovery use case where plugins will call Read.

**Comment by t0yv0** [comment]
I'm wondering if we could contribute here simultaneously as we build some bridge tests to revisit how it all works.
